src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -193,12 +261,25 @@ private static synchronized void loadData() {
           continue;
         }
         ZoneRules zoneRules = zoneId.getRules();
-        // Filter by non-repeating rules
-        if (!zoneRules.isFixedOffset() && !zoneRules.getTransitionRules().isEmpty()) {
-          continue;
-        }
         if (!zoneIdToTable.containsKey(zoneId.getId())) {
-          List<ZoneOffsetTransition> transitions = zoneRules.getTransitions();
+          List<ZoneOffsetTransition> zoneOffsetTransitions = new ArrayList<>(zoneRules.getTransitions());
+          // It is desired to get lastTransitionEpochSecond because some rules don't start until late (e.g. 2007)
+          long lastTransitionEpochSecond = Long.MIN_VALUE;
+          if (!zoneOffsetTransitions.isEmpty()) {
+            long transitionInstant = zoneOffsetTransitions.get(zoneOffsetTransitions.size()-1).getInstant().getEpochSecond();
```

Comment:
It seems like that the Java contract does not guarantee that `zoneRules.getTransitions()` would always return a sorted result. We should not make such an assumption unless it is clearly stated in the contract. I would recommend computing `lastTransitionEpochSecond` after `zoneOffsetTransitions` is sorted as we are going to sort it anyway.